lxd/nsexec.go

```diff
@@ -162,6 +162,18 @@ int manip_file_in_ns(char *rootfs, int pid, char *host, char *container, bool is
 	if (is_put && stat(container, &st) < 0)
 		exists = 0;
 
+	if (is_put) {
+		char *pos = strrchr(container, '/');
+		if (pos) {
+			*pos = 0;
+			if (mkdir_p(container, 0755) < 0) {
```

Comment:
I think this code would need to be moved inside the `!exists` check at [line 185](https://github.com/tych0/lxd/blob/fa2ecdf1c528f26809ba203cf4041c162ccba001/lxd/nsexec.go#L185) after configuring the uid/gid/mode defaults.